nidirect_money_advice_articles/src/EventSubscriber/PostMigrationSubscriber.php

```diff
@@ -0,0 +1,140 @@
+<?php
+
+namespace Drupal\nidirect_money_advice_articles\EventSubscriber;
+
+use Drupal\Core\Database\Database;
+use Drupal\Core\Entity\EntityTypeManagerInterface;
+use Drupal\migrate\Event\MigrateEvents;
+use Drupal\migrate\Event\MigrateImportEvent;
+use Symfony\Component\EventDispatcher\EventSubscriberInterface;
+use Drupal\Core\Logger\LoggerChannelFactory;
+
+/**
+ * Class PostMigrationSubscriber.
+ *
+ * Post Migrate processes.
+ */
+class PostMigrationSubscriber implements EventSubscriberInterface {
+
+  /**
+   * Drupal\Core\Logger\LoggerChannelFactory definition.
+   *
+   * @var \Drupal\Core\Logger\LoggerChannelFactory
+   */
+  protected $logger;
+
+  /**
+   * Stores the entity type manager.
+   *
+   * @var \Drupal\Core\Entity\EntityTypeManagerInterface
+   */
+  protected $entityTypeManager;
+
+  /**
+   * Drupal 8 database connection.
+   *
+   * @var \Drupal\Core\Database\Connection
+   */
+  protected $dbConnDrupal8;
+
+  /**
+   * PostMigrationSubscriber constructor.
+   *
+   * @param \Drupal\Core\Entity\EntityTypeManagerInterface $entity_type_manager
+   *   The Entity manager.
+   * @param \Drupal\Core\Logger\LoggerChannelFactory $logger
+   *   Drupal logger.
+   */
+  public function __construct(EntityTypeManagerInterface $entity_type_manager,
+                              LoggerChannelFactory $logger) {
+    $this->entityTypeManager = $entity_type_manager;
+    $this->logger = $logger->get('nidirect_money_advice_articles');
+    $this->dbConnDrupal8 = Database::getConnection('default', 'default');
+  }
+
+  /**
+   * Get subscribed events.
+   *
+   * @inheritdoc
+   */
+  public static function getSubscribedEvents() {
+    $events[MigrateEvents::POST_IMPORT][] = ['onMigratePostImport'];
+    return $events;
+  }
+
+  /**
+   * Handle post import migration event.
+   *
+   * @param \Drupal\migrate\Event\MigrateImportEvent $event
+   *   The import event object.
+   */
+  public function onMigratePostImport(MigrateImportEvent $event) {
+    $event_id = $event->getMigration()->getBaseId();
+
+    // Only process nodes, nothing else.
+    if ($event_id === 'money_advice_service_rss_articles') {
+      $this->logger->notice($this->processFlags());
+    }
+  }
+
+  /**
+   * Set locked content flag for MAS articles.
+   *
+   * @return string
+   *   Info about the result.
+   * @throws \Exception
+   */
+  protected function processFlags() {
+    // Clear out the flag_counts table so we can reinsert new values.
```

Comment:
This makes me nervous, what of there are articles locked but not part of the MAS, they'll be unlocked.